solr/contrib/ltr/src/java/org/apache/solr/response/transform/LTRFeatureLoggerTransformerFactory.java

```diff
@@ -67,13 +68,26 @@
 
   private String loggingModelName = DEFAULT_LOGGING_MODEL_NAME;
 
+  private LTRThreadModule threadManager = null;
+
   public void setLoggingModelName(String loggingModelName) {
     this.loggingModelName = loggingModelName;
   }
 
   @Override
   public void init(@SuppressWarnings("rawtypes") NamedList args) {
     super.init(args);
+
+    final NamedList threadManagerArgs = LTRThreadModule.extractThreadModuleParams(args);
+    // if and only if there are thread module params then we want a thread module!
+    if (threadManagerArgs.size() > 0) {
+      // create and initialize the new instance
+      threadManager = new LTRThreadModule();
```

Comment:
Any reason not to make LTRThreadModule constructor take in NamedList args as a param in the constructor, and then you dont have to do any of this size > 0 checking and null assignment.  Then just have the constructor call extractThreadModuleParams() and init() internally.
The code for the parser and transformer would just be 1 line then.